### PR TITLE
Add dependency from pipeline to jobs

### PR DESCRIPTION
**Pull request by akm**
In order to start pipeline automatically after some pipelines are finished( or succeeded/failed) , Add an attribute `Dependency` to `Pipeline`.
`Dependency` has `Condition` and `JobIDs`. 

### `Dependency.Condition` 

- `OnFailure`
- `OnSuccess`
- `OnFinished`

### Pipeline status `Pending`

A pipeline is `Pending` until its dependency is satisfied.
After it's satisfied, the pipeline will be `Waiting` or `Reserved`.

### CODE REVIEW

File: src/models/dependency.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package models
+
+import (
+	"golang.org/x/net/context"
+)
+
+type DependencyCondition int
+
+const (
+	OnFailure DependencyCondition = 1 + iota
```
Comment (nagachika) [inline src/models/dependency.go]: Why is the constant incremented from iota?

File: src/models/job.go [hunk 2]
```diff
@@ -63,6 +72,10 @@ func (js JobStatus) Living() bool {
 	return js.IncludedIn(LivingJobStatuses)
 }
 
+func (js JobStatus) Finished() bool {
+	return js.IncludedIn(FinishedJobStatuses)
+}
```
Comment (minimum2scp) [inline src/models/job.go]: Are `IsFinished()` and `Finished()` same function?

File: src/models/dependency.go (hunk 1 above)
Comment (akm) [inline src/models/dependency.go]: @nagachika I use enumeration which starts from 1 to distinguish uninitialized value.
But this doesn't need to distinguish it. I fixed it.

File: src/models/job.go (hunk 2 above)
Comment (akm) [inline src/models/job.go]: @minimum2scp I removed `IsFinished`. Thanks.

**Pull request merged**